portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -151,6 +137,22 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     return blogMessages;
   }
 
+  private List<String> updateTagsToSearch (List<String> tagsToSearch) {
+    UserService userService = UserServiceFactory.getUserService();
+    String email = (String) userService.getCurrentUser().getEmail();
+ 
+    // Check to see if user follows any tags
+    if (LoadFollowedTags.hasFollowedTags(email)) {
```

Comment:
Always think about returning early to avoid deep nesting like:

if (!LoadFollowedTags.hasFollowedTags(email)) {
  return;
}
// Your core logic.